template/src/ontology/Makefile.jinja2

```diff
@@ -219,13 +219,19 @@ RELEASE_ASSETS = \
   $(REPORT_FILES){% endif %}
 
 .PHONY: all_assets
-all_assets: $(ASSETS)
+all_assets: $(ASSETS) check_rdfxml_assets
```

Comment:
This is missing the jinja conditionals?

Reply:
The check rule itself is intended to be unconditional – executed as part of the standard workflow with no opt-out option.

But by default, only _one_ check using the LightRDF parser is enabled. What the `extra_rdfxml_checks` option does is instructing the `check-rdfxml` script to also perform two more checks using the RDFLib and Jena parsers.

The rationale for this behaviour is that:

1) LightRDF is both quite fast and strict, so it doesn’t cost a lot to always run a check with it and it should ensure that any produced RDF/XML file is actually correct. LightRDF is basically high gain, low cost.

2) By contrast, the other parsers are slower (RDFLib in particular is _very_ slow) and don’t appear to be stricter than LightRDF (from some tests I did over the weekend, RDFLib seems to be just as strict as LightRDF, and Jena is laxer), so they don’t offer much. The option to use them is intended for people who want extra assurance that their RDF/XML files can really be consumed by most parsers out there, in the off-chance that they could catch an corner-case error that is somehow missed by LightRDF.

Reply:
Unfortunately I think we need a conditional - this could introduce breaking changes and by design, we should never introduced breaking changes in an ODK update unless we provide a way to switch it off. . I would prefer the whole rdfxml checking pipeline to only be activated if `ensure_valid_rdfxml: TRUE`. This is consistent with `ensure_valid_owl2dl`.

Reply:
OK, makes sense. I would favour the lightrdf check being enabled by default, though. I am concerned almost nobody would turn it on otherwise, which would defeat the whole point of having this check in the first place.

If people find out they have invalid RDF/XML output files after upgrading their ODK instance, and if they decide they don’t care, all they will have to do to restore a “working” pipeline is to change one line in their config file.

Reply:
100% switch on by default!